Source/src/components/ComponentButton.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "Component.h"
+#include "Globals.h"
+#include "ISelectable.h"
+
+namespace Hachiko
+{
+    class GameObject;
+
+    class ComponentTransform2D;
+
+    class ComponentButton : public Component, public ISelectable
+    {
+    public:
+        ComponentButton(GameObject* container);
+        ~ComponentButton() override = default;
+        
+        void Activate() override {}
+
+        static Type GetType();
+
+        void DrawGui() override;
+
+        void Save(JsonFormatterValue j_component) const override;
+        void Load(JsonFormatterValue j_component) override;
+    };
+} // namespace Hachiko
+
+inline Hachiko::Component::Type Hachiko::ComponentButton::GetType()
+{
+    return Type::BUTTON;
+}
```

Comment:
```suggestion
    };
    
    inline Component::Type Hachiko::ComponentButton::GetType()
    {
        return Type::BUTTON;
    }
} // namespace Hachiko
```
Maybe this would be better? I do it this way but I don't remember if we put this specific case to the style doc.

Reply:
Yeah it is already inside the namespace idk why I did that